lib/elixir/src/elixir_clauses.erl

```diff
@@ -186,11 +186,15 @@ expand_with_else(Meta, Opts, E, HasMatch) ->
   form_error(Meta, ?key(E, file), elixir_expand, {missing_option, 'try', [do]});
 'try'(Meta, [{do, _}], E) ->
   form_error(Meta, ?key(E, file), elixir_expand, {missing_option, 'try', ['catch', 'rescue', 'after', 'else']});
-'try'(Meta, [{do, _}, {else, _}], E) ->
-  form_error(Meta, ?key(E, file), ?MODULE, {try_with_only_else_clause, origin(Meta, 'try')});
 'try'(Meta, Opts, E) when not is_list(Opts) ->
   form_error(Meta, ?key(E, file), elixir_expand, {invalid_args, 'try'});
 'try'(Meta, Opts, E) ->
+  case Opts of
+    [{do, _}, {else, _}] ->
+      form_warn(Meta, ?key(E, file), ?MODULE, {try_with_only_else_clause, origin(Meta, 'try')});
```

Comment:
As @josevalim mentioned, it would be nice to include a `TODO` comment to convert this into an error in future Elixir versions :)